pandas/_libs/tslibs/tzconversion.pyx

```diff
@@ -388,14 +390,26 @@ timedelta-like}
                     new_local = val - remaining_mins - 1
 
                 if is_zi:
-                    raise NotImplementedError(
-                        "nonexistent shifting is not implemented with ZoneInfo tzinfos"
-                    )
+                    # use the same construction as in _get_utc_bounds_zoneinfo
+                    pandas_datetime_to_datetimestruct(new_local, creso, &dts)
+                    extra = (dts.ps // 1000) * (pps // 1_000_000_000)
+
+                    dt = datetime_new(dts.year, dts.month, dts.day, dts.hour,
+                                      dts.min, dts.sec, dts.us, None)
 
-                delta_idx = bisect_right_i8(info.tdata, new_local, info.ntrans)
+                    if shift_forward or shift_delta > 0:
+                        dt = dt.replace(tzinfo=tz, fold=1)
```

Comment:
I thought `fold` is only used for ambiguous times not necessarily nonexistent ones?

> In [0, 1]. Used to disambiguate wall times during a repeated interval. (A repeated interval occurs when clocks are rolled back at the end of daylight saving time

Reply:
hmm this may mean that this only handles shift_forward but not shift_backward.  cc @pganssle suggestions on how to handle this?

Reply:
Fold also affects imaginary times. See this: https://pytz-deprecation-shim.readthedocs.io/en/latest/migration.html#semantic-differences-between-is-dst-and-fold

`fold=1` basically means "during a time when it is unclear which of two offsets to use, use the one it's going to be after the ambiguity resolves. `fold=0` resolves to the one it was before the period of ambiguity began.

Reply:
Ah okay thanks for the reference. Guess the stdlib docs could have used clarifying